File: weagle-api/src/main/java/com/weagle/entity/Project.java
package com.weagle.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "project")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Project {

    @Id
    private String id;

    private String name;

    private String description;

    private String strategyId;

    private String ideaId;

    private int progress;

    private String results;

    private double investment;

    private double financialReturn;

    private double productivityIncrease;

    private double costReduction;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: app/src/main/java/com/fiapos/weagle/domain/models/Project.kt
package com.fiapos.weagle.domain.models

import java.time.LocalDate

data class Project(
    val id: String,
    val name: String,
    val description: String,
    val status: ProjectStatus,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val investment: Float,
    val ideaList: List<Idea> = mutableListOf(),
    val ownedBy: String
)